src/Nethermind/Nethermind.Core/BlockToProduce.cs

```diff
@@ -49,5 +49,10 @@ public BlockToProduce(BlockHeader blockHeader, IEnumerable<Transaction> transact
         public BlockToProduce(BlockHeader blockHeader) : base(blockHeader)
         {
         }
+
+        public override Block CreateBlockForProcessing(BlockHeader header) => 
+            new BlockToProduce(header, Transactions, Ommers);
+
+        public override IEnumerable<Transaction> GetTransactions() => Transactions;
```

Comment:
do not assume there will be more implementations, inheritance does not seem necessary?